### PR TITLE
Update docs for Release 4 #no-public-changes

### PR DESCRIPTION
**Pull request by JakeLin**
In this PR, we add some documents for Release 4. We are about to ship release 4 with nice, clean and more understandable animation APIs 💪

### CODE REVIEW

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+## Benefits of upgrading
+- **New promise-like animation APIs:** uses promise-like animation API to replace nested closures syntax.
+
+## Breaking API Changes
+IBAnimatable 4 has introduced new promise-like animation APIs. If you used the old animation APIs in code before, this document can help you identify the most changes for those APIs.
+
+### Animation APIs
+#### Animations in Interface Builder
+If you can configure the animations in Interface Builder. There is no breaking changes at all 🎉
```
Comment (tbaranes) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: if you were configuring? The `can` sounds weird 🤔

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+```swift
+view.animate(.squeezeFade(way: .in, direction: .down))
+    .then(.pop(repeatCount: 1))
+    .then(.shake(repeatCount: 1))
+    .then(.squeeze(way: .in, direction: .down))
+    .then(.wobble(repeatCount: 1))
+    .then(.flip(along: .x))
+    .then(.flip(along: .y))
+    .then(.slideFade(way: .out, direction: .down))
```
Comment (tbaranes) [inline Documentation/IBAnimatable 4.0 Migration Guide.md:72]: Maybe add the completion closure at the end to remember that it still exists?

File: README.md [hunk 3]
```diff
@@ -32,14 +31,20 @@ With `IBAnimatable`, we can design a UI in Interface Builder like what we can do
...
 
-If you are using Xcode 8 with Swift 3, please use the latest tagged 3.x release.
+## Swift version
+### Swift 3 or 3.1
+If you migrate from Swift 2.x, please check out [IBAnimatable 3.0 Migration Guide](Documentation/IBAnimatable 3.0 Migration Guide.md) for more information about how to migrate your project to 3.0. Version 3 follows Swift 3 [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) and contains a lot of breaking changes from version 2.x. 
 
-## Swift 2.2 or 2.3?
+If you are using Xcode 8 with Swift 3, please use the latest tagged 4.x release.
+
+### Swift 2.2 or 2.3?
```
Comment (tbaranes) [inline README.md:43]: I think we can removed that part since we are not supporting it anymore

**Review by tbaranes — COMMENTED** [review]
I did a quick read, it seems good, could be great if an english native can review it too 👍 
There was a few formulation I'm not sure about, but it's maybe due to my english 😬 

Will read it again later! 

Also, we may want to wait for #435 since it introduces breaking change too

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+- iOS 8.0+
+- Xcode 8.0+
+- Swift 3.0+
+
+## Benefits of upgrading
+- **New promise-like animation APIs:** uses promise-like animation API to replace nested closures syntax.
+
+## Breaking API Changes
+IBAnimatable 4 has introduced new promise-like animation APIs. If you used the old animation APIs in code before, this document can help you identify the most changes for those APIs.
```
Comment (SD10) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: "the most changes" -> "most of the changes" or just "most changes"

File: Documentation/IBAnimatable 4.0 Migration Guide.md (hunk 1 above)
Comment (SD10) [reply Documentation/IBAnimatable 4.0 Migration Guide.md]: Yes.
~"If you previously configured the animations in Interface Builder"~
"If your animations are configured in Interface Builder"

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+### Animation APIs
+#### Animations in Interface Builder
+If you can configure the animations in Interface Builder. There is no breaking changes at all 🎉 
+
+#### Animations in code
+
+There is an example how we improve animation APIs. Before version 4, we create chain-able animations like
```
Comment (SD10) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: "This is an example of how we've improved the animation APIs."

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  view.velocity = 2
+  view.pop(repeatCount: 1) {
+    // There is a side effect, the `velocity` for the next animation is also `2`
+  	  view.shake(repeatCount: 1)
+  }
+}
+```
+After we set `velocity` for animation `pop`, we create a side effect to pass the value to the next animation. In this case, which is `shake`.
+
+To fix those issues, we create promise-like animation APIs.
```
Comment (SD10) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: "create" -> "created a"

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+That's nice, clean and more understandable.
+
+#### Animation API Changes
+##### No more completion closures
+As you have seen above, we remove all completion closures. We can use nice syntax `animate`, `delay` and `then` to compose the chain-able animations.
+
+##### Removing individual animation API
+We remove all individual animation API like `slide(.in, direction: .left)`, `pop(repeatCount: 1)` and son on. They are replaced by `animate()` method with `AnimationType` enum like `animate(.slide(way: .in, direction: .left))` and `animate(.pop(repeatCount: 1))`. You can find all changes in [this commit: Update "Predefined Animations" to new animation API](https://github.com/IBAnimatable/IBAnimatable/commit/17e1f9c6c77fea56ac2a47d0fb48fba6c5da972d)
```
Comment (SD10) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: "`pop(repeatCount: 1)` and son on." -> "so on"

File: Documentation/IBAnimatable 4.0 Migration Guide.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+##### Customizing animation in parameters
+Before, to customize the animation, we need to change the view's properties and then call `animate()` method.
```
Comment (SD10) [inline Documentation/IBAnimatable 4.0 Migration Guide.md]: " we need to change" -> "we needed to change"

File: README.md [hunk 9]
```diff
@@ -32,14 +31,20 @@ With `IBAnimatable`, we can design a UI in Interface Builder like what we can do
...
 If you are using Xcode 7.3.1 with Swift 2.2 please use IBAnimatable 2.7. If you are using Xcode 8 with Swift 2.3, please use the latest tagged 2.x release (version 2.8.1). If you find any issue and create a PR for Swift 2.3, please PR to `swift2` branch. When you use Swift 2.3 with IBAnimatable, you may see some issue like `dlopen(IBAnimatable.framework, 1): Symbol not found: __TMVs20_DisabledRangeIndex_`. That's a bug of Xcode 8, please have a look at [Issue - Failed to render and update auto layout status](https://github.com/IBAnimatable/IBAnimatable/issues/349)
 
+**Version 2.8.1 is the last version to support Swift 2.3. Because Xcode 8.3 has stopped supporting Swift 2.3. We also stop the support for Swift 2.*, please migrate to Swift 3.**
```
Comment (SD10) [inline README.md]: "We also stop the support for Swift 2.*" -> "We also stopped supporting Swift 2.*"

File: Documentation/IBAnimatable 4.0 Migration Guide.md (hunk 2 above)
Comment (JakeLin) [reply Documentation/IBAnimatable 4.0 Migration Guide.md:72]: @tbaranes we do support, adding it now.

File: Documentation/IBAnimatable 4.0 Migration Guide.md (hunk 1 above)
Comment (JakeLin) [reply Documentation/IBAnimatable 4.0 Migration Guide.md]: @SD10 thanks, fixed.

File: README.md (hunk 3 above)
Comment (JakeLin) [reply README.md:43]: @tbaranes keep if for another while to let the user know we don't' support Swift 2.*?

**Comment by JakeLin** [comment]
Hey guys, @tbaranes @SD10 I have updated the docs. Please have a second look when you have time, thanks.

**Review by tbaranes — APPROVED** [review]
Just wondering, should we add a section "Others breaking changes" explaining that there also others breaking changes explained in the CHANGELOG?

**Comment by JakeLin** [comment]
@tbaranes Have added `Other breaking changes` in `IBAnimatable 4.0 Migration Guide.md`. Can we merge this PR now? I think we can start the release process after that.

**Comment by tbaranes** [comment]
Ready to go, good job on this @JakeLin 👍

**Pull request merged**